pkg/glanceapi/statefulset.go

```diff
@@ -52,10 +52,11 @@ func StatefulSet(
 	annotations map[string]string,
 	privileged bool,
 ) (*appsv1.StatefulSet, error) {
-	runAsUser := int64(0)
-
+	userID := glance.GlanceUID
+	if privileged {
+		userID = int64(0)
```

Comment:
This is actually used when Cinder is the backend, and the concern is mostly related to the os-brick locks here. I'm wondering if we can still run it as glance user, and let the `kolla` group access the os-brick locks shared by glance and cinder (/cc @eharney )